nox/virtualenv.py

```diff
@@ -85,6 +85,39 @@ def locate_via_py(version):
             return None
 
 
+def locate_using_path_and_version(version):
+    """Check the PATH's python interpreter and return it if the version
+    matches.
+
+    On systems without version-named interpreters and with missing
+    launcher (which is on all Windows Anaconda installations),
+    we search the PATH for a plain "python" interpreter and accept it
+    if its --version matches the specified interpreter version.
+
+    Args:
+        version (str): The desired Python version. Of the form ``X.Y``.
+
+    Returns:
+        Optional[str]: The full executable path for the Python ``version``,
+        if it is found.
+    """
+    if not version:
+        return None
+
+    path_python = py.path.local.sysfind("python")
+    if path_python:
+        try:
+            version_pattern = "^Python {}".format(version)
```

Comment:
Rather than having to use a regex, why not just `platform.python_version()`:

```
$ python2.7 -c 'import platform; print(platform.python_version())'
2.7.15
$ python3.5 -c 'import platform; print(platform.python_version())'
3.5.6
$ python3.6 -c 'import platform; print(platform.python_version())'
3.6.7
$ python3.7 -c 'import platform; print(platform.python_version())'
3.7.1
$ pypy -c 'import platform; print(platform.python_version())'
2.7.13
$ pypy3 -c 'import platform; print(platform.python_version())'
3.5.3
```

Also, if you do go with `python --version`, why resort to a regex when you could just do

```python
prefix = "Python {}.".format(version)  # Note the extra dot
version_string.startswith(prefix)
```